spec/controllers/your_details_controller_spec.rb

```diff
@@ -70,34 +70,30 @@ module Wpcc
 
     describe '#create' do
       context 'success' do
-        it 'stores the form input in a session' do
-          post :create,
-               locale: 'en',
-               your_details_form: {
-                 age: 34,
-                 gender: 'female',
-                 salary: 30_000,
-                 salary_frequency: 'month',
-                 contribution_preference: 'full'
-               }
+        after { Timecop.return }
+
+        it 'stores the form input in a session for 30 minutes' do
+          post_create
 
           expect(session['age']).to eq('34')
           expect(session['gender']).to eq('female')
           expect(session['salary']).to eq('30000')
           expect(session['salary_frequency']).to eq('month')
           expect(session['contribution_preference']).to eq('full')
+
+          Timecop.travel(30.minutes.from_now) do
+            get :new
+            expect(session['age']).to be_nil
```

Comment:
@margOnline I feel that we should set the expectations on the assigned `your_details_form` object. This will provide more confidence that the form object for the new/create action is being created/updated with the right session details.

For example, 
```
it 'sets salary frequency previously added' do
   expect(your_details_form.salary_frequency).to eq('year')
end
```

This test implies the proper functioning of the underlying sessions mechanism.